data-imputation/data_imputation_solution.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+"""data-imputation
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/1ARyd91_CVIlO8H0ncYhNWamQVX5yi1-x
+"""
+
+import pandas as pd
+import numpy as np
+from itertools import groupby
+import random
+
+"""Quiz 1. Handling null values in datasets is important because doing so ensures the quality of our analysis. Namely, we preserve the integrity of the dataset and reduce bias.
```

Comment:
The biggest reason we need to handle null values is because most machine learning algorithms will break if they encounter missing values. The _method_ that we use to impute null values may introduce bias, however, so we need to be careful in choosing which method to use.